src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -4888,13 +4912,12 @@ string Inner()
 			//windows 10 localization concat the xUid Value with the member value (Text, Content, Header etc...)
 			var fullKey = uidName + "/" + memberName;
 
-			if (owner != null && IsAttachedProperty(owner))
+			if (owner != null && IsAttachedProperty(owner, memberName))
 			{
-				var declaringType = GetType(owner.Member.DeclaringType);
+				var declaringType = owner;
 				var nsRaw = declaringType.ContainingNamespace.GetFullName();
-				var ns = nsRaw?.Replace(".", "/");
 				var type = declaringType.Name;
-				fullKey = $"{uidName}/[using:{ns}]{type}/{memberName}";
+				fullKey = $"{uidName}/[using:{nsRaw}]{type}.{memberName}";
```

Comment:
Do we have any tests for when `ContainingNamespace` is the global namespace?

Reply:
In that particular case, the global namespace cannot be used (it's supposed to come from a type that can only be defined from a non-empty namespace).